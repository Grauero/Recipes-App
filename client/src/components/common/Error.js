import React from 'react';
import PropTypes from 'prop-types';

const Error = ({ error }) => (
  <div className="App">
    <h1>An Error Occurred. Try to restart site.</h1>
    {error && error.message && <p>{error.message}</p>}
  </div>
);

Error.propTypes = {
  error: PropTypes.shape({
    message: PropTypes.string
  })
};

Error.defaultProps = {
  error: {}
};

export default Error;
