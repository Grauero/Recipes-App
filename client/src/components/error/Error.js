import React from 'react';
import PropTypes from 'prop-types';

const Error = ({ error }) => <p>{error.message}</p>;

Error.propTypes = {
  error: PropTypes.shape({
    message: PropTypes.string.isRequired
  }).isRequired
};

export default Error;
