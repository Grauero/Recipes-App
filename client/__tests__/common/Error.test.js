import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Error from '../../src/components/common/Error';

it('matches snapshot when error message is provided', () => {
  const props = { error: { message: 'error-message' } };
  const component = shallow(<Error {...props} />);

  expect(toJSON(component)).toMatchSnapshot();
});

it('matches snapshot when error message is NOT provided', () => {
  const component = shallow(<Error />);

  expect(toJSON(component)).toMatchSnapshot();
});
