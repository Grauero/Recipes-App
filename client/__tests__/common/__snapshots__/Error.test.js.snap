// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when error message is NOT provided 1`] = `
<div
  className="App"
>
  <h1>
    An Error Occurred. Try to restart site.
  </h1>
</div>
`;

exports[`matches snapshot when error message is provided 1`] = `
<div
  className="App"
>
  <h1>
    An Error Occurred. Try to restart site.
  </h1>
  <p>
    error-message
  </p>
</div>
`;
